sdks/python/apache_beam/yaml/yaml_transform.py

```diff
@@ -955,10 +955,37 @@ def preprocess_languages(spec):
     else:
       return spec
 
+  def validate_transform_references(spec):
+    if 'transforms' not in spec:
+      return spec
+
+    for transform in spec['transforms']:
+      name = transform.get('name')
+      inputs = transform.get('input')
+      if name is None or inputs is None:
+        continue
+
+      input_values = []
```

Comment:
Let's put this phase after normalize_inputs_outputs to avoid duplicating this logic here.

Reply:
done !